types/react/index.d.ts

```diff
@@ -1743,7 +1743,7 @@ declare namespace React {
     interface HTMLAttributes<T> extends AriaAttributes, DOMAttributes<T> {
         // React-specific Attributes
         defaultChecked?: boolean;
-        defaultValue?: string | number | string[];
+        defaultValue?: string | number | ReadonlyArray<string>;
```

Comment:
It doesn't matter whether the array is readonly or not. It will get casted to a string anyway.

Reply:
It does matter for consumer since if array is mutable consumer can't pass ReadonlyArray but if array is readonly component will consume both mutable and ReadonlyArray.